addons/binding/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -0,0 +1,131 @@
+ */
+public class RokuHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(RokuHandler.class);
+
+    public RokuHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String ipAddress = (String) this.getConfig().get(IP_ADDRESS);
+        Number port = (Number) this.getConfig().get(PORT);
+        if ("REFRESH".equals(command.toFullString())) {
+            updateData();
+        } else {
+            RokuCommands rokuDevice = new RokuCommands(ipAddress, port);
+            try {
+                rokuDevice.generateAction(channelUID);
+            } catch (IOException e) {
+                logger.debug("IOException occurred when attempting action with device '{}'", ipAddress, e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+            // Currently, all of the channels for this binding are implementations of button-presses on the Roku remote.
+            // After a button is "pressed", we'll want to turn it off ("unpress" it), so schedule a job to do so soon.
+            scheduler.schedule(() -> {
+                updateState(channelUID, OnOffType.OFF);
+            }, 1, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing RokuDevice handler '{}'", getThing().getUID());
+        Number refreshInterval = (Number) this.getConfig().get(REFRESH_INTERVAL);
+        scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                updateData();
+            }
+        }, 0, refreshInterval.longValue(), TimeUnit.SECONDS);
+        logger.debug("Polling job scheduled to run every {} sec. for '{}'", refreshInterval, getThing().getUID());
+    }
+
+    private synchronized void updateData() {
+        logger.info("Update Roku data '{}'", getThing().getUID());
+        String ipAddress = (String) this.getConfig().get(IP_ADDRESS);
+        Number port = (Number) this.getConfig().get(PORT);
+        RokuState state = new RokuState(new RokuCommunication(ipAddress, port));
+        try {
+            state.updateDeviceInformation();
+        } catch (IOException e) {
+            logger.debug("Roku device '{}' but is not communicating", ipAddress, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        updatePropertiesFromState(state);
+
+        updateState(CHANNEL_ACTIVE, state.getActive());
+        updateState(CHANNEL_HEADPHONES, new StringType(state.getHeadphonesConnected()));
+        updateState(CHANNEL_ICON, state.getActiveImage());
+        /**
+         * Future Functionality
+         * updateState(CHANNEL_APPBROWSER, state.getApplicationMenu());
+         */
+        if (!"PowerOn".equals(state.getPowerMode())) {
```

Comment:
If you reverse the if you can get rid of the the `not` in general that reads better